### ISSUE TITLE
Conditional move can lead to accidental moves, when opponent moves

### ISSUE THREAD

**Issue by roy7** [online-go/online-go.com#48]
This conversation I think summaries the issue and concludes with a good suggestion.

[10:13]slavka012: Current implementation of conditional moves is horrible
[10:14]slavka012: I like to use it when playing against slow players
[10:14]slavka012: but when they make a move it immediately switches back to game
[10:15]slavka012: if I happen to be making conditional move in the split second that it switches back, I end up making an unintended move in the real game
[10:24]roy7: When using conditional moves maybe you should change settings to the Double Click or Click + Submit option to prevent that :)
[10:36]slavka012: ah, ok.
[10:36]slavka012: not perfect but I can live with that
[10:37]slavka012: keep options page open in second window and switch on and off when needed
[10:37]roy7: Maybe "first move after closing conditional move window" should require click to submit. Or conditional move window shouldn't close automatically at all.
[10:38]roy7: Just reset to new state but still be in conditional move input mode.
[10:38]slavka012: That would be ideal, reset all of the conditional moves, but stay in conditionals, or give me a modal dialog box warning we exit conditionals so it is impossible to make a move

So basically, if you are entering conditional moves and the opponent makes a play, clear the conditional move situation and reset to new board but remain in the conditional move input system until you choose to exit it. Otherwise if you auto-exit moments before clicking, you might make a wrong move to live board by accident.

**Comment by hyperpape** [comment]
Related problem: an attempt to submit a conditional move can turn into a pass. This happened to me a few days ago.

**Comment by Nazgand** [comment]
Since discussion is still needed: It happened to me too. I think that staying in Conditional Move Planner is the correct choice. The conditional move tree should be updated. If you are on the branch that predicted the correct move, then you should barely notice that the opponent moved. Else, if the branch exists on the conditional move tree, you should be moved to the start of that branch. Else is up for debate, but exiting Conditional Move Planner without a warning is a bad idea.

**Comment by MortimerMcMire315** [comment]
It seems like this should be a bit more high-priority than "enhancement". This is a pretty serious interface pitfall - accidental moves can destroy a game if your opponent isn't so gracious as to let you undo.

**Comment by nathanwalther** [comment]
I agree the current behavior is risky. Also analysis and conditional moves are accomplishing similar things with different visualizations of the game tree and different behavior when the other player moves. Personally I like the analysis interface better of the two.

How about making conditional moves part of Analyze Mode? The player can work out a few moves in analysis, then "Save as Conditional Moves" if they like the result of that branch. This could even work when it's your turn, with a confirmation dialog to play the first move with the rest as conditional.
